subsys/fs/ext2/Kconfig

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023 Antmicro
+# SPDX-License-Identifier: Apache-2.0
+
+# Ext2 file system
+
+config FILE_SYSTEM_EXT2
+	bool "Ext2 file system support"
+	depends on FILE_SYSTEM
+	help
+	  Enable Ext2 file system support.
+
+module = EXT2
+module-str = Ext2
+source "subsys/logging/Kconfig.template.log_config"
+
+if FILE_SYSTEM_EXT2
+
+menu "Ext2 file system Settings"
+	visible if FILE_SYSTEM_EXT2
+
+config EXT2_HEAP_SIZE
+	int "Heap size"
+	default 10240
+	help
+	  Memory for allocation of internal structures (e.g. inodes).
```

Comment:
Let me ask if I got everything right.
* `ext2_file` and `ext2_dir` will use common slab which size will be specified by `CONFIG_MAX_FILES`
* `ext2_inode` will use slab for inodes (adjusted to hold enough number of inodes to handle number of files specified with `CONFIG_MAX_FILES`)
* `ext2_disk_dentry` must be allocated on heap because their size may differ (and I will create separate heap for them)
* `ext2_bgroup` and `ext2_disk_superblock` are allocated only once. Because we can mount only single instance of ext2 file system at once I will statically allocate these structures. (And after allowing multiple instances of ext2 they will be allocated from some heap or slab.)

Reply:
Okay, I have updated it.